acceptance_modelisation/toolbox.py

```diff
@@ -30,4 +32,83 @@ def compute_rotation_speed_fov(time_evaluation: Time,
+    i=0
+    mask_allremaining = np.ones(shape=len(all_ra_observations),dtype=bool)
+    while len(ra_observations)>0:
+        i=i+1
+        keywobble='W'+str(i)
+        mask = (angular_separation(ra_observations[0]*u.deg, dec_observations[0]*u.deg,
+                                   ra_observations*u.deg, dec_observations*u.deg) < max_angular_separation*u.deg)
+        mask_2 = (angular_separation(np.mean(ra_observations[mask])*u.deg, np.mean(dec_observations[mask])*u.deg,
+                                     all_ra_observations*u.deg, all_dec_observations*u.deg) < max_angular_separation*u.deg)
+        wobbles_dict[keywobble] = [np.mean(all_ra_observations[mask_2 & mask_allremaining]), np.mean(all_dec_observations[mask_2 & mask_allremaining])]
+        wobbles[mask_2 & mask_allremaining] = keywobble
+        mask_allremaining = mask_allremaining * ~mask_2
+        ra_observations = all_ra_observations[mask_allremaining]
+        dec_observations = all_dec_observations[mask_allremaining]
+
+    print(f"{len(wobbles_dict)} wobbles were found:")
+    for key, value in wobbles_dict.items(): print(f"{key}: {list(np.round(value, 2))}")
+    return wobbles
+
+def plot_coszd_binning(cos_zenith_observations,cos_zenith_bin,bin_center,livetime_observations,min_cut_per_cos_zenith_bin,cos_zenith_binning_method,zd_lim=(55,75),wobble_observations=None,livetime_observations_and_wobble=None):
+    fig,ax=plt.subplots(figsize=(5,5))
+    wobble = (wobble_observations is not None) and (livetime_observations_and_wobble is not None)
+    title = f"Livetime{' per wobble and'*wobble} per cos(zd) bin\n"
+    if wobble:
+        for i,wobble in enumerate(np.unique(np.array(wobble_observations))): 
+            ax.hist(cos_zenith_observations, bins=cos_zenith_bin,weights=livetime_observations_and_wobble[i],alpha=0.6,label=f"{wobble}")
+    else: ax.hist(cos_zenith_observations, bins=cos_zenith_bin,weights=livetime_observations,alpha=0.6)
+
+    new_ticks_coszd=np.concatenate(([np.cos(zd_lim[1]*u.deg)],bin_center,[np.cos(zd_lim[0]*u.deg)]))
+
+    ax.set_xticks(new_ticks_coszd)
+    ax.set_xticklabels(np.round(new_ticks_coszd,2), rotation=45)
+
+    # Create a second x-axis
+    ax2 = ax.twiny()
+    ax2.set_xticks(new_ticks_coszd)
+    ax2.set_xticklabels(np.degrees(np.arccos(new_ticks_coszd)).astype(int), rotation=45)
+
+    # Set labels
+    ax.set_xlabel('cos(zd) bin center')
+    ax.set_ylabel('livetime [s]')
+    ax2.set_xlabel('cos(zd) bin center in zenith [°]')
+
+    xlim=(np.cos(zd_lim[1]*u.deg),np.cos(zd_lim[0]*u.deg))
+    ax.set_xlim(xlim)
+    ax2.set_xlim(xlim)
+    if 'livetime' in cos_zenith_binning_method: ax.hlines([min_cut_per_cos_zenith_bin],xlim[0],xlim[1],ls='-',color='red',label='min livetime',alpha=0.5)
+    
+    ylim=ax.get_ylim()    
+    ax.vlines(cos_zenith_bin,ylim[0],ylim[1],ls=':',color='grey',label='bin edges',alpha=0.5)
+    ax.legend(loc='best')
+
+    plt.suptitle(title)
+    plt.tight_layout()
+    plt.show()
```

Comment:
missing black line

Reply:
It got solved when I removed the plot function